solr/core/src/java/org/apache/solr/cluster/placement/AttributeValues.java

```diff
@@ -65,11 +65,16 @@
   /**
    * For the given node: metric of specific name and registry
    */
-  Optional<Double> getMetric(Node node, String metricName, AttributeFetcher.NodeMetricRegistry registry);
+  Optional<Object> getMetric(Node node, String metricName, AttributeFetcher.NodeMetricRegistry registry);
 
 
   /**
-   * Get a non node related metric of specific scope and name
+   * Get any metric using a fully-qualified metric key.
    */
-  Optional<Double> getMetric(String scope, String metricName);
+  Optional<Object> getNodeMetric(Node node, String metricKey);
+
+  /**
+   *
+   */
+  Optional<CollectionMetrics> getCollectionMetrics(String collectionName);
```

Comment:
We have a lack of symmetry now between node metrics that return mere `Object`s and collection/shard/replica metrics that are defined types (eventually resolving to a `ReplicaMetrics` that can return an object or specific typed metrics - currently only replica size GB).
Maybe we should call something different the "typed" metric interfaces to distinguish between them and the object being eventually returned?
I suggest that rather than calling the interface `CollectionMetrics`, we call it `CollectionMetricContainer` (and similarly for Shard and Replica). That would help me.

Reply:
Hmm... It's clear that this type is not a primitive. Similarly, we're not calling the container for attribute values an `AttributeValuesContainer`. I don't think this change here would actually help in understanding this relationship.